awx_collection/plugins/modules/tower_workflow_launch.py

```diff
@@ -70,97 +95,118 @@
-        if module.check_mode:
-            result['msg'] = "Check mode passed"
-            module.exit_json(**result)
-
-        # We are no ready to run the workflow
-        try:
-            result['job_info'] = tower_cli.get_resource('workflow_job').launch(
-                workflow_job_template=workflow['id'],
-                monitor=False,
-                wait=wait,
-                timeout=timeout,
-                extra_vars=extra_vars
-            )
-            if wait:
-                # If we were waiting for a result we will fail if the workflow failed
-                if result['job_info']['failed']:
-                    result['msg'] = "Workflow execution failed"
-                    module.fail_json(**result)
-                else:
-                    module.exit_json(**result)
-
-            # We were not waiting and there should be no way we can make it here without the workflow fired off so we can return a success
-            module.exit_json(**result)
-
-        except TowerCLIError as e:
-            result['msg'] = "Failed to execute workflow: {0}".format(e)
-            module.fail_json(**result)
+    # Create a datastructure to pass into our job launch
+    post_data = {}
+    for key in optional_args.keys():
+        if optional_args[key]:
+            post_data[key] = optional_args[key]
+
+    # Attempt to look up the related items the user specified (these will fail the module if not found)
+    if inventory:
+        post_data['inventory'] = module.resolve_name_to_id('inventories', inventory)
+
+    # Attempt to look up job_template based on the provided name
+    lookup_data = {'name': name}
+    if organization:
+        lookup_data['organization'] = module.resolve_name_to_id('organizations', organization)
+    workflow_job_template = module.get_one('workflow_job_templates', **{'data': lookup_data})
+
+    if workflow_job_template is None:
+        module.fail_json(msg="Unable to find workflow job template")
+
+    # The API will allow you to submit values to a jb launch that are not prompt on launch.
+    # Therefore, we will test to see if anything is set which is not prompt on launch and fail.
+    check_vars_to_prompts = {
+        'inventory': 'ask_inventory_on_launch',
+        'limit': 'ask_limit_on_launch',
+        'scm_branch': 'ask_scm_branch_on_launch',
+        'extra_vars': 'ask_variables_on_launch',
+    }
+
+    param_errors = []
+    for variable_name in check_vars_to_prompts:
+        if variable_name in post_data and not workflow_job_template[check_vars_to_prompts[variable_name]]:
+            param_errors.append("The field {0} was specified but the workflow job template does not allow for it to be overridden".format(variable_name))
+    if len(param_errors) > 0:
+        module.fail_json(msg="Parameters specified which can not be passed into wotkflow job template, see errors for details", **{'errors': param_errors})
```

Comment:
There is no reason to use `**` notation. It's a lot more clear to just pass `errors=param_errors`. Same for all the times we pass data.